standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/retryhandling/SqlRetryHandler.java

```diff
@@ -123,32 +124,29 @@ public <Result> Result executeWithRetry(SqlRetryCallProperties properties, SqlRe
       if (properties.isLockInternally()) {
         lockInternal();
       }
-      threadLocal.set(new ContextNode<>(threadLocal.get(), new Object()));
+      threadLocal.set(new Object());
```

Comment:
what does it do?

Reply:
In `SqlRetryHandler` the context doesn't hold any information. It's just a marker if we already have an established context or not. The reason it's not a boolean flag that theoretically there can be any level of nesting which we need to keep track of.